### PR TITLE
AESinkPULSE: Check if module-allow-passthrough is loaded and warn if not available

### PR DESCRIPTION
**Pull request by fritsch**
While implementing HD audio support for PA I realized this shiny module-allow-passthrough:

From PA 10 changelog:
New module for prioritizing passthrough streams (module-allow-passthrough)
image

A very useful feature, especially when opening PT might fail.

### CODE REVIEW

File: xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp [hunk 1]
```diff
@@ -280,6 +280,17 @@ struct SinkInfoStruct
   }
 };
 
+struct ModuleInfoStruct
+{
+  pa_threaded_mainloop *mainloop;
+  bool hasAllowPT;
+  ModuleInfoStruct()
+  {
+    mainloop = NULL;
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp]: `nullptr`

File: xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp [hunk 2]
```diff
@@ -385,6 +396,20 @@ static CAEChannelInfo PAChannelToAEChannelMap(const pa_channel_map& channels)
   return info;
 }
 
+static void ModuleInfoCallback(pa_context* c, const pa_module_info *i, int eol, void *userdata)
+{
+  ModuleInfoStruct *mis = (ModuleInfoStruct *)userdata;
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp]: C++ style cast

File: xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp [hunk 3]
```diff
@@ -385,6 +396,20 @@ static CAEChannelInfo PAChannelToAEChannelMap(const pa_channel_map& channels)
   return info;
 }
 
+static void ModuleInfoCallback(pa_context* c, const pa_module_info *i, int eol, void *userdata)
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp]: wrap this function inside `#if PA_CHECK_VERSION(10,0,0)` (unused function warning)

File: xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp (hunk 1 above)
Comment (fritsch) [reply xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp]: In general pulseaudio is a C API. Therefore I decided to keep the coding standards this file already has.

**Comment by fritsch** [comment]
@Rechi could you have another look? I have to keep NULL whenever I call into C functions, which is pulse api to not have to do something like nullptr ? NULL and so on.

Thanks much.

**Comment by fritsch** [comment]
Build failure as usual :-)

**Pull request merged**